pkg/backup/item_backupper.go

```diff
@@ -125,24 +162,23 @@ func (ib *itemBackupper) backupItem(logger logrus.FieldLogger, obj runtime.Unstr
 	if _, exists := ib.backupRequest.BackedUpItems[key]; exists {
 		log.Info("Skipping item because it's already been backed up.")
 		// returning true since this item *is* in the backup, even though we're not backing it up here
-		return true, nil
+		return true, itemFiles, nil
 	}
 	ib.backupRequest.BackedUpItems[key] = struct{}{}
-
 	log.Info("Backing up item")
 
-	log.Debug("Executing pre hooks")
-	if err := ib.itemHookHandler.HandleHooks(log, groupResource, obj, ib.backupRequest.ResourceHooks, hook.PhasePre); err != nil {
-		return false, err
-	}
-
 	var (
 		backupErrs []error
 		pod        *corev1api.Pod
 		pvbVolumes []string
 	)
 
-	if groupResource == kuberesource.Pods {
+	log.Debug("Executing pre hooks")
+	if err := ib.itemHookHandler.HandleHooks(log, groupResource, obj, ib.backupRequest.ResourceHooks, hook.PhasePre); err != nil {
+		return false, itemFiles, err
+	}
+
+	if !finalize && groupResource == kuberesource.Pods {
```

Comment:
So there will be problem if we backup data in the `Finalization` phase?

So the `Finalization` phase was introduced to handle the `itemsToUpdate` list, and make sure they are backed up, I was thinking we should handle these items as how we did in the regular backup process.

We should avoid the case that the` itemsToUpdate` are handled differently in many ways in terms of backup.
If there's a very strong reason for that, let's make sure we document the difference in the flow and the reason behind such difference.

Reply:
See my general comment at the end. I think I'm closer to agreement with you here now, at least for pods/pvcs. I think I will probably be able to remove this finalize special case.